roles/libvirt_manager/tasks/add_static_ip_script.yml

```diff
@@ -0,0 +1,81 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Prepares the virtual machine boot image. Though the generation is
+# straight forward, we need to ensure support exists for static IP address
+# configuration.
+#
+# Multiple images are created for a single vm type in the case of static IP
+# address. Also, the index is used to determine the next usable IP address.
+
+
+- name: Generate the static address configuration script
+  vars:
+    _ip_address: >-
+      {{
+        vm_data.value.ip_address.address |
+        ansible.utils.next_nth_usable(vm_id)
```

Comment:
sanity check i guess the vm_id here is sequential like 1/2/3 etc?

Reply:
forgot to add link in case useful for others (I had to go find that ) https://docs.ansible.com/ansible/latest/collections/ansible/utils/next_nth_usable_filter.html

Reply:
The sequence is `0..N` driven by `range` and assigned as `vm_id` by `loop_control.index_var`